olegbetsun/booking_up_for_beauty.go

```diff
@@ -0,0 +1,52 @@
+package booking
+
+import (
+	"time"
+	"strconv"
+	"fmt"
+	)
+
+func ParseDate(layout string, date string) time.Time{
+	t,err := time.Parse(layout, date)
+	if err != nil {
+		fmt.Printf("%v \n", err)
+	}
+	return t
+}
+
+
+// Schedule returns a time.Time from a string containing a date
+func Schedule(date string) time.Time {
+	layout := "1/02/2006 15:04:05"
+	t:= ParseDate(layout, date)
+	return t
+}
+
+// HasPassed returns whether a date has passed
+func HasPassed(date string) bool {
+	layout := "January 2, 2006 15:04:05"
+	t := ParseDate(layout, date)
+	return t.Before(time.Now())
+}
+
+// IsAfternoonAppointment returns whether a time is in the afternoon
+func IsAfternoonAppointment(date string) bool {
+	layout := "Monday, January 2, 2006 15:04:05"
+	t:= ParseDate(layout, date)
+	noon := time.Date(t.Year(), t.Month(), t.Day(), 12, 0, 0, t.Nanosecond(), t.Location())
+	eve := time.Date(t.Year(), t.Month(), t.Day(), 17, 0, 0, t.Nanosecond(), t.Location())
+	return t.After(noon) && t.Before(eve)
+}
+
+// Description returns a formatted string of the appointment time
+func Description(date string) string {
+	layout := "1/2/2006 15:04:05"
+	t:= ParseDate(layout, date)
+	return "You have an appointment on "+t.Weekday().String()+", "+t.Month().String()+" "+strconv.Itoa(t.Day())+", "+strconv.Itoa(t.Year())+", at "+strconv.Itoa(t.Hour())+":"+strconv.Itoa(t.Minute())+"."
```

Comment:
time has a [Format function](https://pkg.go.dev/time#Time.Format) can simplify the creation of this string
```suggestion
	return "You have an appointment on "+t.Format("Monday, January 2, 2006, at 15:04")
```

Reply:
Thank you, Sam!

Reply:
I'd also use fmt.Sprintf to generate your string instead of doing those conversions. Something like:
```
fmt.Sprintf("You have an appointment on %s %d", t.Month(), t.Day())
```

Reply:
I read fmt.Sprintf is not very effective in doing that int to string conversion and that's better to use strconv.Itoa() instead. Here are some refs https://stackoverflow.com/questions/19223277/how-to-convert-uint8-to-string/62725637#62725637

Reply:
Pretty much:
```
goos: linux
goarch: amd64
pkg: test
cpu: Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
BenchmarkCastWithItoa
BenchmarkCastWithItoa-12                	377316148	         2.857 ns/op
BenchmarkCastWithSprintf
BenchmarkCastWithSprintf-12             	16763394	        71.39 ns/op
BenchmarkCastWithItoa10numbers
BenchmarkCastWithItoa10numbers-12       	 9758014	       125.9 ns/op
BenchmarkCastWithSprintf10numbers
BenchmarkCastWithSprintf10numbers-12    	 3066861	       399.1 ns/op
PASS
ok  	test	5.648s
```